plugins/flytekit-pydantic/flytekitplugins/pydantic/deserialization.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Callable, Dict, Iterator, List, Type, TypeVar, Union
+
+import pydantic
+
+from flytekit.types import directory, file
+
+from . import object_store
+
+# this field is used by pydantic to get the validator method
+PYDANTIC_VALIDATOR_METHOD_NAME = pydantic.BaseModel.__get_validators__.__name__
+PythonType = TypeVar("PythonType")  # target type of the deserialization
+Serializable = TypeVar("Serializable")  # flyte object type
+
+
+def set_validators_on_supported_flyte_types() -> None:
+    """
+    Sets the validator on the pydantic model for the
+    type that is being serialized/deserialized
+    """
```

Comment:
```suggestion
    """Set the validator on the pydantic model for the type that is being (de-)serialized.
    """
```

Nitpick (non-blocking): [PEP257](https://peps.python.org/pep-0257/) says 

> The docstring is a phrase ending in a period. It prescribes the function or method’s effect as a command (“Do this”, “Return that”), not as a description; e.g. don’t write “Returns the pathname …”.